src/sqlfluff/core/parser/lexer.py

```diff
@@ -42,125 +76,121 @@ def __init__(
         self,
         name,
         template,
-        target_seg_class,
-        subdivide=None,
+        segment_class,
+        subdivider=None,
         trim_post_subdivide=None,
-        *args,
-        **kwargs
+        segment_kwargs=None,
     ):
         self.name = name
         self.template = template
-        self.target_seg_class = target_seg_class
-        self.subdivide = subdivide
+        self.segment_class = segment_class
+        self.subdivider = subdivider
         self.trim_post_subdivide = trim_post_subdivide
+        self.segment_kwargs = segment_kwargs or {}
 
-    def _match(self, forward_string):
-        """The private match function. Just look for a single character match."""
-        if forward_string[0] == self.template:
-            return forward_string[0]
+    def __repr__(self):
+        return f"<{self.__class__.__name__}: {self.name}>"
+
+    def _match(self, forward_string: str) -> Optional[LexedElement]:
+        """The private match function. Just look for a literal string."""
+        if forward_string.startswith(self.template):
+            return LexedElement(self.template, self)
         else:
             return None
 
-    def _trim(self, matched, start_pos):
+    def search(self, forward_string: str) -> Optional[Tuple[int, int]]:
+        """Use string methods to find a substring."""
+        if self.template in forward_string:
```

Comment:
Performance nitpick: Here we are searching for the string twice, first with `in`, then with `find`. If `find()` doesn't find the string, it returns `-1`, so we could potentially streamline this.